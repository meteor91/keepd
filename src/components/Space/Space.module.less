.space {
  display: flex;
  gap: 8px;

  &-start {
    justify-content: start;
  }
  &-end {
    justify-content: end;
  }
  &-center {
    justify-content: center;
  }
  &-space-around {
    justify-content: space-around;
  }
  &-space-between {
    justify-content: space-between;
  }
  &-space-evenly {
    justify-content: space-evenly;
  }
}