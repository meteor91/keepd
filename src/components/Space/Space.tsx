import React, { Children } from 'react';
import clsx from 'clsx';
import styles from './Space.module.less';

type SpaceJustify = 'start' | 'end' | 'center' | 'space-around' | 'space-between' | 'space-evenly';

interface SpaceProps {
    children?: React.ReactNode;
    justify?: SpaceJustify;
}

export const Space: React.FC<SpaceProps> = (props) => {
    const { children, justify } = props;

    const nodes = Children.toArray(children).map((child, index) => {
        // TODO подумать о key
        const key = `space-item-${index}`;
        return (
            <div key={key}>
                {child}
            </div>
        );
    });

    return (
        <div className={clsx(styles.space, styles[`space-${justify}`])}>
            {nodes}
        </div>
    );
};