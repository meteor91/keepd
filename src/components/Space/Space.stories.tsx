import React from 'react';
import type { Meta, StoryObj } from '@storybook/react';

import { Space } from './Space';
import { Button } from '../Button';

const meta: Meta<typeof Space> = {
    title: 'Space',
    component: Space,
    argTypes: {
        justify: {
            control: { type: 'radio' },
            options: ['start', 'end', 'center', 'space-around', 'space-between', 'space-evenly'],
        },
    },
};

export default meta;

type Story = StoryObj<typeof Space>;

export const Basic: Story = {
    render: (args) => {
        return (
            <Space justify={args.justify}>
                <Button label="btn-1"/>
                <Button label="btn-2"/>
            </Space>
        );
    },
};
