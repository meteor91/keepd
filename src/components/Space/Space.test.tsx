import React from 'react';
import { render } from '@testing-library/react';

import { Space } from './Space';

describe('Space', () => {
    it('should render correctly', () => {
        const { container } = render(
            <Space justify="end">
                <div>test 1</div>
                <div>test 2</div>
            </Space>,
        );

        expect(container.firstChild).toMatchSnapshot();
    });
});