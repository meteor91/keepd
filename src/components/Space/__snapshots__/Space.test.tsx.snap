// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Space should render correctly 1`] = `
<div
  class="space space-end"
>
  <div>
    <div>
      test 1
    </div>
  </div>
  <div>
    <div>
      test 2
    </div>
  </div>
</div>
`;
